modules/javafx.controls/src/main/java/javafx/scene/control/ControlUtils.java

```diff
@@ -166,19 +167,26 @@ public static <S> void updateSelectedIndices(MultipleSelectionModelBase<S> sm, b
             // ensuring that the selectedIndices bitset is correctly updated.
 
             sm.startAtomic();
-            final List<Integer> removed = c.getRemoved().stream()
+
+            final List<Integer> removed = new ArrayList<>(c.getRemovedSize());
+            c.getRemoved().stream()
                     .mapToInt(TablePositionBase::getRow)
                     .distinct()
                     .filter(removeRowFilter)
-                    .boxed()
-                    .peek(sm.selectedIndices::clear)
-                    .collect(Collectors.toList());
+                    .forEach(row -> {
+                        removed.add(row);
+                        sm.selectedIndices.clear(row);
+                    });
 
-            final int addedSize = (int)c.getAddedSubList().stream()
+            final int[] addedSize = new int[1];
+            c.getAddedSubList().stream()
                     .mapToInt(TablePositionBase::getRow)
                     .distinct()
-                    .peek(sm.selectedIndices::set)
-                    .count();
+                    .forEach(row -> {
```

Comment:
Similar comment as above. Here if we do not use `forEach()` on streams, we can also avoid using array of size one for keeping count as well right?

Reply:
We'd need to iterate twice as well (select index & count), with forEach it's just once.